src/platform_impl/linux/x11/mod.rs

```diff
@@ -442,7 +442,7 @@ impl<T: 'static> EventLoop<T> {
             // Reset the internal state for the loop as we start running to
             // ensure consistent behaviour in case the loop runs and exits more
             // than once.
-            self.set_control_flow(ControlFlow::Poll);
```

Comment:
Same here, the reason is to kick start the first entrance to the loop, the rest should be fine.

Reply:
Moved to https://github.com/rust-windowing/winit/pull/3107, as it seems that this is an issue unrelated to this PR.